docs/sidebar.reference.yml

```diff
@@ -28,12 +28,13 @@ subsection:
     directory: contextual
     subsection:
```

Comment:
This file is exactly the old `sidebar.yml` for only reference related sections. 
Additions here are mostly due to diverging state of `https://github.com/scala/scala3/tree/language-reference-stable` that was not sync with main